packages/nimble-components/src/table/tests/table-selection.spec.ts

```diff
@@ -648,76 +774,201 @@ describe('Table row selection', () => {
+                        clickModifiers: {},
+                        expectedSelection: ['1'],
+                        fireEvent: true
+                    },
+                    {
+                        name: 'CTRL + clicking a selected row with multiple rows selected deselects the clicked row, keeps others selected, and emits an event',
+                        initialSelection: ['1', '2'],
+                        rowToClick: 1,
+                        clickModifiers: { ctrlKey: true },
+                        expectedSelection: ['2'],
+                        fireEvent: true
+                    }
+                ];
+                const focused: string[] = [];
+                const disabled: string[] = [];
+                for (const configuration of configurations) {
+                    const specType = getSpecTypeByNamedList(
+                        configuration,
+                        focused,
+                        disabled
                     );
-                    const emittedIds = getEmittedRecordIdsFromSpy(
-                        selectionChangeListener.spy
-                    );
-                    expect(emittedIds).toEqual(expectedSelection);
-                });
+                    // eslint-disable-next-line @typescript-eslint/no-loop-func
+                    specType(configuration.name, async () => {
+                        await element.setSelectedRecordIds(
+                            configuration.initialSelection
+                        );
+                        await pageObject.clickRow(
+                            configuration.rowToClick,
+                            configuration.clickModifiers
+                        );
 
-                it('clicking a row with one different row previously selected selects the clicked row and deselects the other row and emits an event', async () => {
-                    await element.setSelectedRecordIds(['1']);
-                    await pageObject.clickRow(0);
+                        const currentSelection = await element.getSelectedRecordIds();
+                        expect(currentSelection).toEqual(
+                            jasmine.arrayWithExactContents(
+                                configuration.expectedSelection
+                            )
+                        );
+                        if (configuration.fireEvent) {
+                            expect(
+                                selectionChangeListener.spy
+                            ).toHaveBeenCalledTimes(1);
+                            const emittedIds = getEmittedRecordIdsFromSpy(
+                                selectionChangeListener.spy
+                            );
+                            expect(emittedIds).toEqual(
+                                jasmine.arrayWithExactContents(
+                                    configuration.expectedSelection
+                                )
+                            );
```

Comment:
<suggestion?> Would it be helpful to wrap at least the contents of this `if` statement in a helper function? I believe this implementation is at least duplicated.

Reply:
Yea keep the expects in the tests and also avoid conditional code paths for running expects like described above. I'd advocate for splitting up the tests into positive case for firing events and negative case.